src/component/parser/json/converter/ProcessConverter.ts

```diff
@@ -104,26 +106,29 @@ export default class ProcessConverter extends AbstractConverter<void> {
     this.buildAssociationFlows(process[FlowKind.ASSOCIATION_FLOW]);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private buildFlowNodeBpmnElements(processId: string, bpmnElements: Array<any> | any, kind: ShapeBpmnElementKind): void {
+  private buildFlowNodeBpmnElements(processId: string, bpmnElements: Array<FlowNode> | FlowNode, kind: ShapeBpmnElementKind): void {
     ensureIsArray(bpmnElements).forEach(bpmnElement => {
       let shapeBpmnElement;
 
       if (ShapeUtil.isEvent(kind)) {
         shapeBpmnElement = this.buildShapeBpmnEvent(bpmnElement, kind as BpmnEventKind, processId);
       } else if (ShapeUtil.isSubProcess(kind)) {
         shapeBpmnElement = this.buildShapeBpmnSubProcess(bpmnElement, processId);
+      } else if (kind === ShapeBpmnElementKind.TEXT_ANNOTATION) {
+        const name = 'text' in bpmnElement ? bpmnElement.text.toString() : undefined;
+        shapeBpmnElement = new ShapeBpmnElement(bpmnElement.id, name, kind, processId);
       } else {
-        const name = kind === ShapeBpmnElementKind.TEXT_ANNOTATION ? bpmnElement.text : bpmnElement.name;
-        shapeBpmnElement = new ShapeBpmnElement(bpmnElement.id, name, kind, processId, bpmnElement.instantiate);
+        const name = 'name' in bpmnElement ? bpmnElement.name : undefined;
+        const instantiate = 'instantiate' in bpmnElement ? bpmnElement.instantiate : undefined;
+        shapeBpmnElement = new ShapeBpmnElement(bpmnElement.id, name, kind, processId, instantiate);
```

Comment:
If we don't use **any** as type, there is an error, because all the types don't have all the properties.

So, we can keep as it was before, or we can have it, or we can separate all flownode convertions in dedicated converters.

Reply:
ok I understand now, I have also done some test locally to see what kind of errors the TS compiler generates.
I don't think we gain something on the code readability here, the code is now harder to read now IMHO